libs/x-content/src/main/java/org/elasticsearch/common/xcontent/ObjectParser.java

```diff
@@ -78,14 +78,63 @@
         };
     }
 
+    private interface UnknownFieldParser<Value, Context> {
+
+        void acceptUnknownField(String parserName, String field, XContentLocation location, XContentParser parser,
+                                Value value, Context context) throws IOException;
+    }
+
+    private static <Value, Context> UnknownFieldParser<Value, Context> ignoreUnknown() {
+      return (n, f, p, x, v, c) -> x.skipChildren();
```

Comment:
Any reason you did not use `(n, f, l, p, v, c)` instead?

Reply:
I think that's a leftover from various attempts at refactoring, will fix.